executor/api/grpc/kfserving/client.go

```diff
@@ -0,0 +1,183 @@
+	ctx = grpc2.AddMetadataToOutgoingGrpcContext(ctx, meta)
+	var resp *proto.ModelInferResponse
+	switch v := msg.GetPayload().(type) {
+	case *proto.ModelInferRequest:
+		resp, err = grpcClient.ModelInfer(ctx, v, s.callOptions...)
+	default:
+		return nil, errors.Errorf("Invalid type %v", v)
+	}
+	if err != nil {
+		return nil, err
+	}
+	resPayload := payload.ProtoPayload{Msg: resp}
+	return &resPayload, nil
+}
+
+func (s *KFServingGrpcClient) TransformInput(ctx context.Context, modelName string, host string, port int32, msg payload.SeldonPayload, meta map[string][]string) (payload.SeldonPayload, error) {
+	panic("implement me")
+}
+
+func (s *KFServingGrpcClient) Route(ctx context.Context, modelName string, host string, port int32, msg payload.SeldonPayload, meta map[string][]string) (int, error) {
+	panic("implement me")
+}
+
+func (s *KFServingGrpcClient) Combine(ctx context.Context, modelName string, host string, port int32, msgs []payload.SeldonPayload, meta map[string][]string) (payload.SeldonPayload, error) {
+	panic("implement me")
+}
+
+func (s *KFServingGrpcClient) TransformOutput(ctx context.Context, modelName string, host string, port int32, msg payload.SeldonPayload, meta map[string][]string) (payload.SeldonPayload, error) {
+	panic("implement me")
+}
+
+func (s *KFServingGrpcClient) Feedback(ctx context.Context, modelName string, host string, port int32, msg payload.SeldonPayload, meta map[string][]string) (payload.SeldonPayload, error) {
+	panic("implement me")
+}
+
+func (s *KFServingGrpcClient) Chain(ctx context.Context, modelName string, msg payload.SeldonPayload) (payload.SeldonPayload, error) {
+	switch v := msg.GetPayload().(type) {
+	case *proto.ModelInferRequest:
+		s.Log.Info("Identity chain")
+		return msg, nil
+	case *proto.ModelInferResponse:
+		s.Log.Info("Chain!")
+		inputTensors := make([]*proto.ModelInferRequest_InferInputTensor, len(v.Outputs))
+		for _, oTensor := range v.Outputs {
+			inputTensor := &proto.ModelInferRequest_InferInputTensor{
+				Name:       oTensor.Name,
+				Datatype:   oTensor.Datatype,
+				Shape:      oTensor.Shape,
+				Parameters: oTensor.Parameters,
+				Contents:   oTensor.Contents,
+			}
+			inputTensors = append(inputTensors, inputTensor)
+		}
+		pr := proto.ModelInferRequest{
+			Inputs: inputTensors,
+		}
+		msg2 := payload.ProtoPayload{Msg: &pr}
+		return &msg2, nil
+	default:
+		return nil, errors.Errorf("Invalid type %v", v)
+	}
+}
+
+func (s *KFServingGrpcClient) Status(ctx context.Context, modelName string, host string, port int32, msg payload.SeldonPayload, meta map[string][]string) (payload.SeldonPayload, error) {
+	conn, err := s.getConnection(host, port, modelName)
+	if err != nil {
+		return nil, err
+	}
+	grpcClient := proto.NewGRPCInferenceServiceClient(conn)
+	ctx = grpc2.AddMetadataToOutgoingGrpcContext(ctx, meta)
+	var resp *proto.ModelReadyResponse
+	switch v := msg.GetPayload().(type) {
+	case *proto.ModelReadyRequest:
```

Comment:
Do we also need to expose the liveness endpoint?

Reply:
Is there a v2 grpc message for this?

Reply:
I think I misunderstood. This seems to be only status of a model, so no liveness!